Lib/test/test_math.py

```diff
@@ -2087,6 +2087,24 @@ def test_nextafter(self):
         self.assertTrue(math.isnan(math.nextafter(1.0, NAN)))
         self.assertTrue(math.isnan(math.nextafter(NAN, NAN)))
 
+    @requires_IEEE_754
+    def test_ulp(self):
+        self.assertEqual(math.ulp(0.0),
+                         sys.float_info.min * sys.float_info.epsilon)
+        self.assertEqual(math.ulp(1.0), sys.float_info.epsilon)
+        self.assertEqual(math.ulp(2.0 ** 52), 1.0)
```

Comment:
Maybe use `sys.float_info.radix ** (sys.float_info.mant_dig-1)`?

Reply:
Tests are hardcoded for binary64. Is it really worth it to make tests generic for other implementations of foat? I'm quite sure that many tests are hardcoded for binary64, I mean, not only test_float.

Reply:
I'd stick a @requires_IEEE754 on this and just make the tests specific to IEEE 754.

Reply:
`2.0 ** 52` is unsafe here, since it relies on the platform libm having an accurate pow, and in general that's not safe. You could use `float(2**52)` or `ldexp(1.0, 52)` instead as a safer alternative.